src/main/java/org/broadinstitute/hellbender/tools/exome/hmm/CopyNumberTriStateHiddenMarkovModel.java

```diff
@@ -0,0 +1,354 @@
+
+                //return MathUtils.logSumLog(new double[] {logF + logQ, log1MinusF + log1Minus2P});
+            } else if (currentState == nextState) {
+                // Event extension transition prob.:
+                return MathUtils.logSumLog(new double[] {logF + log1MinusQ, log1MinusF + logP});
+            } else {
+                // Event "sign switch" (del. <--> dup.) transition prob.:
+                return log1MinusF + logP;
+            }
+        }
+    }
+
+    /**
+     * Calculate the distance between two targets.
+     * <p>
+     * If the target provided do not contain intervals, the distance is set to 0 in order to revert
+     * to the non-distance dependent model described in Fromer et al. paper
+     * <a href="http://www.ncbi.nlm.nih.gov/pmc/articles/PMC3484655/table/tbl1/">Table 1</a>.
+     * </p>
+     * <p>
+     * If both targets map to different chromosomes then we return {@link Double#POSITIVE_INFINITY}.
+     * </p>
+     * @param fromTarget the previous target.
+     * @param toTarget the next target.
+     * @return any values between 0 and {@link Double#POSITIVE_INFINITY}.
+     * @throws NullPointerException if any of the targets is {@code null}.
+     */
+    @VisibleForTesting
+    double calculateDistance(final Target fromTarget, final Target toTarget) {
+        final SimpleInterval fromInterval = fromTarget.getInterval();
+        final SimpleInterval toInterval = toTarget.getInterval();
+        if (fromInterval == null || toInterval == null) {
+            return 0.0;
+        } else if (!fromInterval.getContig().equals(toInterval.getContig())) {
+            return Double.POSITIVE_INFINITY;
+        } else {
+            final double distance = toInterval.getStart() - fromInterval.getEnd() - 1;
+            return distance > 0 ? distance : Math.max(0, fromInterval.getStart() - toInterval.getEnd() - 1);
+        }
+    }
+
+
+    @Override
+    public double logEmissionProbability(final Double data,
+                                         final CopyNumberTriState state,
+                                         final Target target) {
+        final NormalDistribution distribution = stateDistribution(Utils.nonNull(state));
+        return distribution.logDensity(data);
+    }
+
+    /**
+     * Returns the appropriate emission distribution given the copy-number state.
+     * @param state the requested state.
+     * @return never {@code null.}
+     */
+    private NormalDistribution stateDistribution(final CopyNumberTriState state) {
+        if (state == CopyNumberTriState.NEUTRAL) {
+            return NEUTRAL_DISTRIBUTION;
+        } else if (state == CopyNumberTriState.DELETION) {
+            return deletionDistribution;
+        } else {
+            return duplicationDistribution;
+        }
+    }
+
+    @VisibleForTesting
+    public Double randomDatum(final CopyNumberTriState state, final Random rdn) {
```

Comment:
If it's `@VisibleForTesting` can't you give it no access modifier to default to package protected instead of making it `public`?